### ISSUE TITLE
[ENH] Work Items for `ptf-v2`

### ISSUE THREAD

**Issue by phoeenniixx** [sktime/pytorch-forecasting#1974]
We already have a beta version released for `ptf-v2`, but still a lot of work is to be done. 
This issue aims to collect and track the pending work items and improvements that need to be discussed and addressed.

Work Items (not in any specific priority):

- [ ] `predict` function and `_pkg` wrapper (as mentioned in #1870 and sktime/enhancement-proposals/pull/46) implementation
- [ ] How to handle `categorical` variables ?
- [ ] Add `scalers` for preprocessing
- [ ] Add more `train_test_split` strategies
- [ ] Add `label_encoders` to D1 Layer?
- [ ] Add support for datasets which cannot fit in memory
- [ ] Move the v1 models to v2 (should be of low priority)

**Comment by phoeenniixx** [comment]
FYI @fkiraly, @agobbifbk, @PranavBhatP 
Please feel free to add more tasks that I may have missed out

**Comment by phoeenniixx** [comment]
Also, I was thinking about the `datamodules`: 
Right now, we have 2 - one for `ptf` native models and one for `tslib`, so I was thinking, maybe moving forward we should also add a Base Class for these data modules as well?

**Comment by fkiraly** [comment]
Preliminary thoughts on priorities. I would advise three parallel tracks:

1. v2 feature parity, with priority in decreasing order:

* `predict` interface
* scalers
* categorical inputs and label encoder

2. adding more models - v1 and/or v2

incl writing an extension template

I think this is important to:

* validate the v2 API as it progresses
* validate the v1 testing framework (and with it v2)
* encourage new contributors

Noting that we will likely have to revisit v2 estimators as we are changing API.

3. attempt full integration with dsipts - using the 2nd party integration pattern as between sktime and prophetverse

**Comment by agobbifbk** [comment]
My two cents for the numerical scalers and not-in-memory data. We can leverage on `partial_fit` method of most of the scalers (unfortunately not for categorical variable) without reinventing the wheel 
https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html#sklearn.preprocessing.StandardScaler.partial_fit

**Comment by ojuschugh1** [comment]
Hi Team, I hope you are doing well. I have choosen to pick on  Add label_encoders to D1 Layer task, i have started the implementation , will open the PR tommorrow.

Thanks
